diff --git a/Source/WebKit/PlatformWPE.cmake b/Source/WebKit/PlatformWPE.cmake
index 92825501798b..be960eda81a4 100644
--- a/Source/WebKit/PlatformWPE.cmake
+++ b/Source/WebKit/PlatformWPE.cmake
@@ -276,7 +276,7 @@ list(APPEND WebKit_DEPENDENCIES
     webkitwpe-forwarding-headers
 )
 
-if (${GI_VERSION} VERSION_GREATER_EQUAL 1.79.2)
+if (GI_VERSION VERSION_GREATER_EQUAL 1.79.2)
     set(USE_GI_FINISH_FUNC_ANNOTATION 1)
 else ()
     set(USE_GI_FINISH_FUNC_ANNOTATION 0)

